generator/src/main/java/com/scottlogic/deg/generator/walker/ReductiveDecisionTreeWalker.java

```diff
@@ -93,28 +82,27 @@ public ReductiveDecisionTreeWalker(
         FixFieldStrategy fixFieldStrategy){
 
         //reduce the tree based on the fields that are now fixed
-        ReductiveConstraintNode reducedNode = this.treeReducer.reduce(constraintNode, reductiveState);
-        if (reducedNode == null){
-            //a field has been fixed, but is contradictory, i.e. it has invalidated the tree
-            //yielding an empty stream will cause back-tracking
+        Combined<ConstraintNode> reducedNode = this.treeReducer.pruneConstraintNode(constraintNode, reductiveState.getLastFixedField());
```

Comment:
`Combined` is a bit confusing here, could this be better named, see comment on `Combined` type